apps/framework-cli-e2e/test/cli.test.ts

```diff
@@ -0,0 +1,212 @@
+    } catch (error) {
+      console.error("Error stopping Docker containers:", error);
+    }
+
+    removeTestProj();
+  });
+
+  it("should return the dummy version in debug build", async () => {
+    const { stdout } = await execAsync(`"${CLI_PATH}" --version`);
+    const version = stdout.trim();
+    const expectedVersion = "moose-cli 0.0.1";
+    expect(version).to.equal(expectedVersion);
+  });
+
+  it("should init a project, install dependencies, run dev command, send a request, and stop it", async function () {
+    this.timeout(120_000); // 2 minutes
+
+    if (fs.existsSync(TEST_PROJECT_DIR)) {
+      removeTestProj();
+    }
+
+    console.log("Initializing project...");
+    await execAsync(
+      `"${CLI_PATH}" init my-moose-app ts --location "${TEST_PROJECT_DIR}"`,
+    );
+
+    console.log("Updating package.json to use local moose-lib...");
+    const packageJsonPath = path.join(TEST_PROJECT_DIR, "package.json");
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf-8"));
+
+    packageJson.dependencies["@514labs/moose"] = `file:${MOOSE_LIB_PATH}`;
+
+    fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2));
+
+    console.log("Installing dependencies...");
+    await new Promise<void>((resolve, reject) => {
+      const npmInstall = spawn("npm", ["install"], {
+        stdio: "inherit",
+        cwd: TEST_PROJECT_DIR,
+      });
+      npmInstall.on("close", (code) => {
+        console.log(`npm install exited with code ${code}`);
+        code === 0
+          ? resolve()
+          : reject(new Error(`npm install failed with code ${code}`));
+      });
+    });
+
+    console.log("Starting dev server...");
+    devProcess = spawn(CLI_PATH, ["dev"], {
+      stdio: "pipe",
+      cwd: TEST_PROJECT_DIR,
+    });
+
+    await new Promise<void>((resolve, reject) => {
+      let serverStarted = false;
+      devProcess!.stdout?.on("data", async (data) => {
+        const output = data.toString();
+        console.log("Dev server output:", output);
+
+        if (
+          !serverStarted &&
+          output.includes(
+            "Your local development server is running at: http://localhost:4000/ingest",
+          )
+        ) {
+          resolve();
+          serverStarted = true;
+        }
+      });
+
+      devProcess!.stderr?.on("data", (data) => {
+        console.error("Dev server stderr:", data.toString());
+      });
+
+      devProcess!.on("exit", (code) => {
+        console.log(`Dev process exited with code ${code}`);
+        expect(code).to.equal(0);
+      });
+
+      (async () => {
+        await setTimeoutAsync(60_000);
+        if (devProcess!.killed) return;
+        console.error("Dev server did not start or complete in time");
+        devProcess!.kill("SIGINT");
+        reject(new Error("Dev server timeout"));
+      })();
+    });
+
+    console.log("Server started, waiting before sending test request...");
+
+    await setTimeoutAsync(2000);
```

Comment:
Would hitting `http://localhost:4000/health` in a loop until success (with a timeout) be more accurate?

Reply:
Actually we don't need the wait.